### ISSUE TITLE
rsync failed with --relative (implied by --files-from)

### ISSUE THREAD

**Issue by huangyingw** [WayneD/rsync#399]
when use with files-from option, it will automatically append $HOME directory to the target directory, so result in a failure:
"/home/myusername/~/workspace/azure-test" No such file or directory (2)

here is the command I used:
```
rsync -aHv --force --progress --files-from rsync.files . sb:~/workspace/azure-test/ 
```

these are the output:

```
rsync -aHv --force --progress --files-from rsync.files . sb:~/workspace/azure-test/ 
Warning: Permanently added '10.2.214.53' (ED25519) to the list of known hosts.
building file list ... 
2 files to consider
rsync: [Receiver] mkdir "/home/myusername/~/workspace/azure-test" failed: No such file or directory (2)
rsync error: error in file IO (code 11) at main.c(789) [Receiver=3.2.7-54-g0db8a943]
```

however, if I remove the files-from option, then it works.
```
rsync -aHv --force --progress . sb:~/workspace/azure-test/ 
```

the content of rsync.file:
```
cat rsync.files
./run.sh
```

**Comment by WayneD** [comment]
I've committed a change that avoids the extra tilde escaping when it's a destination arg in a --relative transfer.

You can avoid the issue in your current rsync by getting rid of `~/` (`sb:workspace/azure-test/`) since remote connections default to the home dir.

**Comment by huangyingw** [comment]
great thanks, I saw your commit, this one:
    537499408948f4e3dd048a861eb58f211e4546f8
right?

I am planning to compile the latest master code. 

Which side should I compile and build, the sender side, receiver side, or both?

So, once I compile and build, this command should work, right?

```
rsync -aHv --force --progress --files-from rsync.files . sb:~/workspace/azure-test/ 
```

**Comment by WayneD** [comment]
The bug affects the client side.  In the command you specified, the client is operating as the sender.

**Comment by huangyingw** [comment]
just found that it is very difficult to build/compile on M1 Chip Mac...
I will try again later.
or maybe I need to wait for your next release...
currently, the version available for homebrew is 3.2.7, ping me, when the next release version is available.

**Comment by huangyingw** [comment]
great thanks to your answer in 
    https://github.com/WayneD/rsync/discussions/404

I have successfully compiled rsync in my m1 chip mac, and verified this issue as fixed.
